### DISCUSSION TITLE
Substack refused to let PageAssist work

### DISCUSSION THREAD

**Discussion by BradKML** [n4ze3m/page-assist#86] [Q&A]
Maybe sites like Substack have an anti-scraping function that does JS-based rendering, can PageAssist take the article after it has been rendered in the browser and do the embedding that way, rather than recalling HTTP request (I think that is how it is currently done)?

**Comment by n4ze3m** [comment]
Yes, I am currently adding a similar feature where you can select a specific component on the site, rather than the entire site, to chat with

**Reply by BradKML** [reply]
What if I want the entire site, but they are in fact weird in their page loading? Can they just load directly from the browser rendering?

**Reply by n4ze3m** [reply]
I just tested the Substack website. The scripting is working fine, but it embedded a lot of unwanted tags, which is why the rag is not working well. I will add a separate parser for Substack websites

**Reply by BradKML** [reply]
Yeah is this an issue of article extraction? If so could you help and evaluate tools like Trafilatura? https://github.com/adbar/trafilatura
Also I am not sure if the extraction benchmark is considered "correct" now, maybe there is a way out with this? https://github.com/scrapinghub/article-extraction-benchmark/issues/3

**Reply by n4ze3m** [reply]
It is Python code, and the Page Assist extension is JavaScript. But anyway, thanks for the suggestions! I will check the code and improve the Page Assist scrapping code 😃😃

**Reply by BradKML** [reply]
I think the benchmark test list includes libraries made with JS? (just that Trafilatura is the best performer in the game but "readability" comes in multiple languages)
